### PR TITLE
Add documentation for dialogue manager

### LINKED ISSUE #239: Add documentation for dialogue manager
Create a new page for DM in the documentation. It should contain information on the following modules:

- Dialogue state tracking
- Dialogue policies

A link to this page should be added to `README.md`.

### PR DESCRIPTION
**Pull request by NoB0**
Fixes #239

### CODE REVIEW

File: docs/source/dialogue_manager.rst [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+---------------
+
+This component decides what is the next action of the system based on the current dialogue state, i.e., it generates the next dialogue act of the system.
+It defines the flow of the conversation, i.e., what steps an agent must take at every stage. 
+MovieBot supports two types of dialogue policies: rule-based and neural-based.
+
+Rule-based Dialogue Policy
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Based on a set of pre-defined rules and the current dialogue state, this policy decides the next action of the system.
```
Comment (kbalog) [inline docs/source/dialogue_manager.rst:24]: Sounds repetitive. Perhaps simplify to "The next system action is determined based on a set of pre-defined rules, given the current dialogue state."

**Review by kbalog — CHANGES_REQUESTED** [review]

File: docs/source/dialogue_manager.rst (hunk 1 above)
Comment (NoB0) [reply docs/source/dialogue_manager.rst]: Replaced by suggestion.

File: docs/source/dialogue_manager.rst [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+The :py:class:`DialogueManager <moviebot.dialogue_manager.dialogue_manager>` tracks the dialogue state and decides what action the system should take.
+It comprises a dialogue state tracker and a dialogue policy.
+
+
+Dialogue State Tracker
+----------------------
+
+This component is responsible for keeping track of the dialogue state and dialogue context based on the dialogue acts of the agent and the user.
+In the current implementation, :py:class:`DialogueStateTracker <moviebot.dialogue_manager.dialogue_state_tracker>` is rule-based.
```
[the remark this answers is not in the document]
Comment (NoB0) [reply docs/source/dialogue_manager.rst]: I removed this sentence and added a short description of the dialogue state.

**Review by kbalog — APPROVED** [review]
LGTM

**Pull request merged**